### DISCUSSION TITLE
Can I contribute?

### DISCUSSION THREAD

**Discussion by colhany** [gusmanb/logicanalyzer#61] [Q&A] [answered]
Hello
First of all I want to thank the author for this incredible project, I recently found it and I find it fascinating.
Now my question is, I've always wanted to contribute to open source hardware but haven't found any projects that grabbed my interest, so can I contribute to the hardware of this project? I'm thinking about creating better PCBs with the microcontrollers on-board instead of a carrier PCB for a dev board, is that possible? If yes, do I need to work with a specific ECAD tool, or whatever I like?
Thank you

**Comment by gusmanb** [comment, marked as answer]
Hi!

Thanks a lot for the offering.

The thing is that is something I evaluated at the beginning of the project and discarded it immediately. The spirit of the project is to have something that anyone with very basic skills can assemble but still be useful enough to be a decent tool. Soldering a Pico board into the pcb is very easy, but having to solder all the components that conform a Pico can be really difficult, mostly the RP2040, the other IC's can be obtained in SMD packages big enough to not be too challenging, but the RP204 being a QFN56 would make the assembly too difficult for a lot of users. Also, there would need to be multiple PCB's to maintain the WiFi functionality which would make it even more difficult.

In any case, which are your skills? There are three main improvements that I want to do but I had some health issues recently again and I'm not able to do it right now so maybe you can help in those.

These are: Real USB coms, burst mode and oscilloscope mode.

If you are comfortable programming in C then the real usb may be a good area to help, right now the device uses the USB port as a serial port and uses stdio functions to send the data, this is a lot slower than what it could be and I woult like to make this a real USB device using LibUSB so it can do much faster transfers, this would also even allow to increase the sampling count as it would be possible to transfer the buffer as it's being captured and until the PIO "catches" the USB it could transfer more data.

The second area is the burst mode, I'm implementing this to be able to leverage as much as it could the memory usage. The device only has 128Kb of buffer so the capture size is tiny when you want to capture large comms, but usually these capture sessions contain many many wasted samples, two devices talking among them usually have a lot of "silence" space, so the idea is to be able to capture only bursts of data, when the trigger condition happens a burst of n samples is captured and the device rearms the trigger again to capture another burst, in this way you end with a much more meaningful set of samples using the available buffer.

Finally, the oscilloscope mode is an expansion card that would convert the device into a two channel 20Mhz oscilloscope, the idea is to use two 100Mhz 8 bit ADCs and use these to convert the device into an oscilloscope. The ADCs that I would like to use are the AD9283-100, I already got two of them to start the development before getting sick, but there I would need help, analog electronics for such high frequencies fall out of my comfort zone so if anyone could help me designing the analog input stage that would be great.

So, do you feel to be able to help with any of these? Maybe if you're more into the electronics side help for the oscilloscope analog stage would be great, what do you think?

Cheers!

**Reply by colhany** [reply]
Dear Agustín,
Thank you very much for the quick and detailed reply.
I understand your vision about making this project easy to put together by anybody, but there are also some people who like to solder and tinker with SMD components who I think would like my idea. I'm not suggesting it as a replacement but rather as a second option.
Also, there are many cheap assembly service providers nowadays like JLCPCB which can assembly the whole PCB. In any case, it's your choice whether you would like to proceed with something like this or not.
As for the improvements you want to make, I'm just starting in embedded software so it's not really one of my strong skills yet. However, the oscilloscope extension seems very attractive to me and in fact, I've always though about working on something like this, so I'd be very happy to help. However, I might not be able to start right away due to my exams, maybe a couple of weeks and I can start if that's possible.
Also please let me know if you want me to use a specific ECAD tool, and I'd be really grateful if you give me any material that you already collected as a start.
I have worked with faster signals for a while so I think I might be able to help with something like this, and as I said, this is an area of interest for me, so I'll also be learning and researching more.
Thank you and good luck!
